### ISSUE TITLE
Argument dim=-1 doesn't work for torch.cross

### ISSUE THREAD

**Issue by you74674** [pytorch/pytorch#17229]
## 🐛 Bug

The argument dim=-1 doesn't work for torch.cross, despite that it is the default argument described in document.

## To Reproduce

```
>>> import torch
>>> a = torch.FloatTensor([[1, 1, 1], [2, 2, 2], [3, 3, 3]])
>>> b = torch.FloatTensor([[4, 4, 4], [5, 5, 5], [6, 6, 6]])
>>> torch.cross(a[0], b[0])
tensor([0., 0., 0.])
>>> torch.cross(a, b, dim=1)
tensor([[0., 0., 0.],
        [0., 0., 0.],
        [0., 0., 0.]])
>>> torch.cross(a, b, dim=0)
tensor([[-3., -3., -3.],
        [ 6.,  6.,  6.],
        [-3., -3., -3.]])
>>> torch.cross(a, b)
tensor([[-3., -3., -3.],
        [ 6.,  6.,  6.],
        [-3., -3., -3.]])
>>> torch.cross(a, b, dim=-1)
tensor([[-3., -3., -3.],
        [ 6.,  6.,  6.],
        [-3., -3., -3.]])
>>> torch.cross(a[:2], b[:2])
tensor([[0., 0., 0.],
        [0., 0., 0.]])
```
## Expected behavior

torch.cross should handle dim=-1.
That is, torch.cross(a, b, dim=-1) should give:
tensor([[0., 0., 0.],
        [0., 0., 0.],
        [0., 0., 0.]])

## Environment

PyTorch version: 1.0.1.post2
Is debug build: No
CUDA used to build PyTorch: 9.0.176

OS: Ubuntu 16.04.5 LTS
GCC version: (Ubuntu 6.5.0-2ubuntu1~16.04) 6.5.0 20181026
CMake version: version 3.14.0-rc1

Python version: 3.6
Is CUDA available: Yes
CUDA runtime version: 10.0.130
GPU models and configuration: GPU 0: GeForce GTX 1080
Nvidia driver version: 410.79
cuDNN version: Probably one of the following:
/usr/local/cuda-10.0/targets/x86_64-linux/lib/libcudnn.so.5
/usr/local/cuda-10.0/targets/x86_64-linux/lib/libcudnn.so.7.3.1

Versions of relevant libraries:
[pip3] numpy==1.14.1
[pip3] numpy-quaternion==2019.2.13.5.15.10
[pip3] torch==1.0.1.post2
[pip3] torchvision==0.2.1
[conda] Could not collect

## Additional context
According to the code here, it only search for the first dimension with size 3 without handling minus dimension.
https://github.com/pytorch/pytorch/blob/73d7ecd18398cf121a9a2ff2faad302d429bef3e/aten/src/TH/generic/THTensorMoreMath.cpp#L487-L501
Besides the bug, the behavior is strange to me, although it matches the document("If dim is not given, it defaults to the first dimension found with the size 3.")
I think it should search from the last dimension instead of the first dimension, since most of the time the first dimension is batch size.

**Comment by HapeMask** [comment]
I also ran into this issue when I was explicitly passing dim=-1 expecting cross() to use the last axis. It was using the last axis except when my batch size (which changes each iteration) was exactly 3 in which case it would take the cross product over the first axis and give the wrong result.

**Comment by fmassa** [comment]
#17582 fix this
